File: examples/typescript/testing/assertions/assert-wait.ts
import { withDurableExecution, DurableContext } from "@aws/durable-execution-sdk-js";
import { LocalDurableTestRunner, OperationStatus } from "@aws/durable-execution-sdk-js-testing";
import { ExecutionStatus, OperationType } from "@aws-sdk/client-lambda";

const handler = withDurableExecution(async (event: unknown, context: DurableContext) => {
  await context.wait("my-wait", { seconds: 30 });
  return "done";
});

let runner: LocalDurableTestRunner;

beforeAll(async () => {
  await LocalDurableTestRunner.setupTestEnvironment({ skipTime: true });
});

afterAll(async () => {
  await LocalDurableTestRunner.teardownTestEnvironment();
});

beforeEach(() => {
  runner = new LocalDurableTestRunner({ handlerFunction: handler });
});

it("asserts on a wait operation", async () => {
  const result = await runner.run();

  expect(result.getStatus()).toBe(ExecutionStatus.SUCCEEDED);

  const wait = runner.getOperation("my-wait");
  expect(wait.getType()).toBe(OperationType.WAIT);
  expect(wait.getStatus()).toBe(OperationStatus.SUCCEEDED);
  expect(wait.getWaitDetails()?.waitSeconds).toBe(30);
  expect(wait.getWaitDetails()?.scheduledEndTimestamp).toBeInstanceOf(Date);
});

File: examples/python/testing/assertions/assert-wait.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution
from aws_durable_execution_sdk_python.config import Duration
from aws_durable_execution_sdk_python.execution import InvocationStatus
from aws_durable_execution_sdk_python.lambda_service import OperationStatus, OperationType
from aws_durable_execution_sdk_python_testing.runner import DurableFunctionTestRunner


@durable_execution
def handler(event, context: DurableContext) -> str:
    context.wait(Duration.from_seconds(30), name="my-wait")
    return "done"


def test_asserts_on_wait_operation():
    runner = DurableFunctionTestRunner(handler=handler)
    with runner:
        result = runner.run(timeout=10)

    assert result.status is InvocationStatus.SUCCEEDED

    wait = result.get_wait("my-wait")
    assert wait.status is OperationStatus.SUCCEEDED
    assert wait.scheduled_end_timestamp is not None
